services/client/config.go

```diff
@@ -0,0 +1,17 @@
+package client
+
+import "time"
+
+// Configuration is the config for service discovery
+type Configuration struct {
+	InitTimeout *time.Duration `yaml:"initTimeout"`
+}
+
+// NewOptions creates an Option
+func (cfg Configuration) NewOptions() Options {
+	opts := NewOptions()
+	if cfg.InitTimeout != nil {
```

Comment:
In that case why not just make `InitTimeout` a `time.Duration` and don't call `SetInitTimeout` if `cfg.InitTimeout == 0`?